### PR TITLE
Indexing nested documents

### PR DESCRIPTION
**Pull request by efagerberg**
This PR contains a small change that adds the ability to index child documents. This is helpful for queries that join together conditions on children and parent documents.

### CODE REVIEW

File: pysolr.py [hunk 1]
```diff
@@ -817,6 +817,11 @@ def _build_doc(self, doc, boost=None, fieldUpdates=None):
                 if self._is_null_value(bit):
                     continue
 
+                if key == '_doc':
+                    child = self._build_doc(bit, boost)
+                    doc_elem.append (child)
```
Comment (acdha) [inline pysolr.py]: The an extra space before the parentheses

**Comment by efagerberg** [comment]
@acdha thanks for the review I fixed the extra space and added a new test for the `_doc` path.

**Comment by efagerberg** [comment]
Is there anything else I need to do here?

**Comment by efagerberg** [comment]
@acdha do you know when this will get approved/introduced? Would like to depreciate our fork so we can keep up to date.

File: setup.py [hunk 2]
```diff
@@ -31,7 +31,7 @@
     ],
     extras_require={
         'solrcloud': [
-    	    'kazoo==2.2'
```
Comment (acdha) [inline setup.py]: I'd like to avoid unrelated changes in this branch

File: tests/test_client.py [hunk 3]
```diff
@@ -544,6 +544,37 @@ def test__build_doc(self):
...
+            'title': 'Example doc ☃ 1',
+            'price': 12.59,
+            'popularity': 10,
+            '_doc': sub_docs
+        }
+        doc_xml = force_unicode(ElementTree.tostring(self.solr._build_doc(doc), encoding='utf-8'))
+        self.assertTrue('<field name="title">Example doc ☃ 1</field>' in doc_xml)
+        self.assertTrue('<field name="id">doc_1</field>' in doc_xml)
+        self.assertTrue('<field name="title">Example sub doc ☃ 1</field>' in doc_xml)
+        self.assertTrue('<field name="id">sub_doc_1</field>' in doc_xml)
```
Comment (acdha) [inline tests/test_client.py]: I've been wondering whether this should actually confirm the structure – i.e. does it seem worth parsing the XML to confirm that the sub doc is contained within the parent document?

Comment (efagerberg) [inline tests/test_client.py]: Is your suggestion to not compare the xml as string but to iterate through the xml elements and confirm each tag and value?

Comment (acdha) [inline tests/test_client.py]: That's what I was thinking – something like an xpath query to find `doc_1` and then confirm that it contains `sub_doc_1`.

Comment (acdha) [inline tests/test_client.py]: Thanks!

File: tests/test_client.py [hunk 4]
```diff
@@ -544,6 +544,37 @@ def test__build_doc(self):
...
+        }
+        doc_xml = self.solr._build_doc(doc)
+        self.assertEqual(doc_xml.find(".//*[@name='id']").text, doc['id'])
+
+        children_docs = doc_xml.findall('doc')
+        self.assertEqual(len(children_docs), len(sub_docs))
+
+        self.assertEqual(children_docs[0].find(".//*[@name='id']").text, sub_docs[0]['id'])
+        self.assertEqual(children_docs[1].find(".//*[@name='id']").text, sub_docs[1]['id'])
+
```
Comment (efagerberg) [inline tests/test_client.py]: @acdha here is the new test approach lmkwyt.

Comment (efagerberg) [inline tests/test_client.py]: womp womp, looks like the tests failed on some python versions :(

Comment (efagerberg) [inline tests/test_client.py]: Ok back to passing again.

**Comment by efagerberg** [comment]
Hello @acdha is there anything else that is required here?

**Comment by efagerberg** [comment]
Sorry to spam you @acdha would just really like to introduce this change in so we can get off of our fork and keep up to date.

**Comment by yooper** [comment]
Please @acdha , can you commit this PR to the project.
👍

**Comment by efagerberg** [comment]
Thank you
